rfcs/0013-changes-lifecyle.md

```diff
@@ -0,0 +1,348 @@
+
+```
+test.toml
+test.feature_a.toml
+```
+
+To avoid duplications `test.feature_a.toml` can inherit properties from `test.toml`.
+And some new properties can be created to allow the configuration of the compiler:
+
+```
+[environment_variables]
+SOME_VAR = "1"
+
+[forc]
+cli = "--experimental feature_a,feature_b"
+```
+
+The same strategy can be used for snapshot tests. We can use multiple `snapshot.toml` and create different
+snapshots using the file suffix. So `snapshot.feature_a.toml` will generate `snapshot@feature_a.snap`.
+
+# Drawbacks
+
+[drawbacks]: #drawbacks
+
+This will increase the complexity of the compiler. Not all flags used to compile end up in the JSON ABI, or other outputs. Which can make reproducibility harder.
+
+# Rationale and alternatives
+
+[rationale-and-alternatives]: #rationale-and-alternatives
+
+There are only two other alternatives, which we consider the worst options:
+
+1. Keep experimental features in branches;
+2. Keep experimental features behind compiler flags (conditional compilation);
+
+The first one normally creates more problems than it solves. "Integration hell" becomes a reality sooner than later.
+The second does not decrease the complexity of the code, and it decreases the testability of features, given that they will not be in the "release binary".
+
+# Prior art
+
+[prior-art]: #prior-art
+
+These are the sources and stacks used as motivation for this RFC.
+
+## How `rustc` does it
+
+https://rustc-dev-guide.rust-lang.org/implementing_new_features.html#stability-in-code
+
+1. Open a tracking issue;
+2. Pick a name for the feature gate;
+3. Add the feature gate to the compiler code;
+4. If the feature gate is not set, you should either maintain the pre-feature behavior or raise an error, depending on what makes sense;
+5. Add a test to ensure the feature cannot be used without a feature gate;
+6. Add a section to the unstable book;
+7. Write a lot of tests for the new feature;
+8. Get your PR reviewed and land it.
+9. Add a section to the unstable book;
+
+## changeset
+
+https://github.com/changesets/changesets
+
+## keepachangelog
+
+Changes will categorized following https://keepachangelog.com/en/1.1.0/
+
+# Unresolved questions
+
+[unresolved-questions]: #unresolved-questions
+
+## Where should we save "release notes"?
+
+1. In files inside the repo? - Saving in files sounds like the best approach, but if we use the same file, all PRs will conflict. To avoid this we can do like `changeset` and use random names.
+2. In GitHub PR descriptions? - This is the easiest approach, but it can be cumbersome to recover these messages.
+3. In git commit messages? - Given that the commit message is "created" when the PR is merged, there is no way to guarantee that "release notes" will exist or be "parseable" when we need them.
+   - https://git-cliff.org/
+
+## Should new warnings be considered breaking changes?
```

Comment:
I'd say breaking for patches, but not for minor releases. At least this is how rust typically works with clippy warnings etc.